framework/tlvf/src/include/tlvf/WSC/AttrList.h

```diff
@@ -0,0 +1,139 @@
+/* SPDX-License-Identifier: BSD-2-Clause-Patent
+ *
+ * Copyright (c) 2019 Tomer Eliyahu (Intel Corporation)
+ *
+ * This code is subject to the terms of the BSD+Patent license.
+ * See LICENSE file for more details.
+ */
+
+#ifndef _TLVF_WSC_ATTR_LIST_H_
+#define _TLVF_WSC_ATTR_LIST_H_
+
+#include <memory>
+#include <tlvf/ClassList.h>
+#include <tlvf/WSC/WSC_Attributes.h>
+#include <tlvf/ieee_1905_1/tlvWsc.h>
+#include <tlvf/tlvflogging.h>
+
+namespace WSC {
+
+typedef struct sWscAttrHeader {
+    uint16_t type;
+    uint16_t length;
+    uint8_t *data() { return ((uint8_t *)this + sizeof(*this)); }
+} __attribute__((packed)) sWscAttrHeader;
+
+/**
+ * @brief wsc configuration
+ * 
+ */
+struct config {
+    eWscMessageType msg_type;
+    uint8_t uuid_e[WSC_UUID_LENGTH];
+    uint8_t uuid_r[WSC_UUID_LENGTH];
+    sMacAddr mac;
+    uint8_t enrollee_nonce[WSC_NONCE_LENGTH];
+    uint8_t registrar_nonce[WSC_NONCE_LENGTH];
+    uint8_t pub_key[WSC_PUBLIC_KEY_LENGTH];
+    uint16_t encr_type_flags;
+    uint16_t auth_type_flags;
+    std::string manufacturer;
+    std::string model_name;
+    std::string model_number;
+    std::string serial_number;
+    uint16_t primary_dev_type_id;
+    std::string device_name;
+    eWscRfBands bands;
+    std::vector<uint8_t> encrypted_settings;
+    uint8_t iv[WSC_ENCRYPTED_SETTINGS_IV_LENGTH];
+};
+
+class AttrList : public ClassList {
+protected:
+    AttrList(uint8_t *buff, size_t buff_len, bool parse) : ClassList(buff, buff_len, parse) {}
+
+public:
+    virtual ~AttrList()                  = default;
+    virtual eWscMessageType type()       = 0;
```

Comment:
Why is this a pure-virtual method and not say a simple getter method?
Won't it be easier to pass the Type in the constructor and have the base class simply return it?
Anyhow, getter method should probably be defined as const (e.g. `virtual eWscMessageType type() const = 0;`)

Reply:
I guess it could be a simple getter, just like the other common attributes getters defined in the AttrList class.